tests/conftest.py

```diff
@@ -941,7 +945,7 @@ def html_index_with_onetime_server(
     """
 
     class InDirectoryServer(http.server.ThreadingHTTPServer):
-        def finish_request(self, request: Any, client_address: Any) -> None:
+        def finish_request(self: "Self", request: Any, client_address: Any) -> None:
```

Comment:
Why Mypy requires this change I don’t understand

Reply:
Without it the call below it:

```python
self.RequestHandlerClass(
    request,
    client_address,
    self,
    directory=str(html_index_for_packages),  # type: ignore[call-arg]
)
```

Causes the mypy error:

```
tests/conftest.py:952: error: Argument 3 has incompatible type
"InDirectoryServer"; expected "Self"  [arg-type]
                    self,
                    ^~~~
```

I did not investigate why mypy now requires `self` in this case to be type hinted.

Reply:
Yeah I kind of understand the reason Mypy needs it, but it really shouldn’t need it. It could be a bug.